### ISSUE TITLE
Use `sklearn`'s OneHotEncoder instead of `rdt`

### ISSUE THREAD

**Issue by katxiao** [sdv-dev/SDMetrics#170]
### Problem Description

Replace `rdt.transformers.OneHotEncoder` with `sklearn.preprocessing.OneHotEncoder` in the following metrics:

```
LogisticDetection,
SVCDetection,
LogisticParentChildDetection
SVCParentChildDetection
```

### Expected behavior

External metric behavior should not change.
